python/cudf/cudf/core/udf/utils.py

```diff
@@ -254,8 +263,14 @@ def _get_input_args_from_frame(fr):
 
 
 def _return_arr_from_dtype(dt, size):
+    extensionty = masked_array_types.get(dt)
+    if extensionty:
+        return rmm.DeviceBuffer(size=size * extensionty.return_type.size_bytes)
     return cp.empty(size, dtype=dt)
 
 
 def _post_process_output_col(col, retty):
+    getter = output_col_getters.get(retty)
+    if getter:
+        col = getter(col)
```

Comment:
This can walrus too. Or alternatively, you could do
```suggestion
    col = output_col_getters.get(retty, lambda x: x)(col)
```
Whatever option you like best, the current code is fine too if you don't find the alternatives as readable.